sgx-hasher/src/lib.rs

```diff
@@ -79,6 +79,55 @@ impl Hasher {
     }
 }
 
+fn bn_to_rsanum_inner(bn: bn::BigNum) -> Result<[u8; 384], openssl::error::ErrorStack> {
+    let mut array = [0u8; 384];
+    array.copy_from_slice(&bn.to_vec());
+    array.reverse();
+    Ok(array)
+}
+
+fn bn_to_u32(value: &bn::BigNum) -> Result<u32, openssl::error::ErrorStack> {
+    let mut buf = u32::default().to_ne_bytes(); // Zero
+    value
+        .to_slice(&mut buf)
+        .or(Err(openssl::error::ErrorStack::get()))?;
+    Ok(u32::from_be_bytes(buf))
+}
+
+pub fn sign(
+    author: sig::Author,
+    contents: sig::Contents,
+    key: rsa::Rsa<pkey::Private>,
+) -> Result<sig::Signature, openssl::error::ErrorStack> {
```

Comment:
I think `std::io::Error` might have been correct here since we need to also return something like `std::io::ErrorKind::InvalidInput` when the integer conversion fails. Sorry for misleading you.